index.js

```diff
@@ -527,8 +527,8 @@ function normalizeArguments(url, opts) {
 		'user-agent': `${pkg.name}/${pkg.version} (https://github.com/sindresorhus/got)`
 	}, headers);
 
-	if (opts.decompress) {
-		opts.headers['accept-encoding'] = 'gzip,deflate';
+	if (opts.decompress && opts.headers['accept-encoding'] == null) {
```

Comment:
This breaks existing functionality. `opts.headers['accept-encoding']` will always evaluate to false due to L522. This means the default headers will never be added.

You'd either need to fix the check, e.g. `if (opts.decompress && !is.undefined(opts.headers['accept-encoding'])` or update the assignment to keep existing values: `opts.headers['accept-encoding'] = opts.headers['accept-encoding'] || 'gzip, deflate';`.

Reply:
Aren't you basically saying that `== null` isn't allowed and Pieter should instead use `is.undefined()`? I don't see how it would break existing functionality.

Reply:
Line 531 is unreachable. The change needs the respect an existing header value, but it's preventing the default from being applied entirely.

Reply:
@brandon93s I think the confusion is the use of `== null`, which actually returns `true` if the value is `undefined` as well (not the `==` rather than `===`). That should result in the line being reachable and executing as expected.

In our internal code style policies this is the idiomatic way of doing such checks, but if you prefer another code style using `is.undefined`, `hasOwnProperty` or the `||` approach I'm happy to do either one :)

Reply:
And I see the test error is actually a code style error where it's instructing not to use `== null`, so I guess I just answered my own question ;)